src/main/java/net/countercraft/movecraft/combat/features/tracking/DamageTracking.java

```diff
@@ -10,22 +10,21 @@
 import net.countercraft.movecraft.events.CraftSinkEvent;
 import org.bukkit.Bukkit;
 import org.bukkit.configuration.file.FileConfiguration;
+import org.bukkit.entity.Player;
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.EventPriority;
 import org.bukkit.event.Listener;
 import org.jetbrains.annotations.NotNull;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
 
 public class DamageTracking implements Listener {
     public static boolean EnableFireballTracking = false;
     public static boolean EnableTNTTracking = true;
     public static boolean EnableTorpedoTracking = false;
     public static int DamageTimeout = 300;
     private final Map<PlayerCraft, List<DamageRecord>> damageRecords = new HashMap<>();
+    private static final Map<Player, Boolean> hideCredMap = new WeakHashMap<>();
```

Comment:
What is the purpose of making this static?  There is already only one instance of the `DmaageTracking` class, so making this static vs a per-instance variable has no functional difference.  We make the config flags static so they can be accessed regardless of if the instance is created or not.

Reply:
I've noticed why this is.  Instead of doing this, let's revert this to a non-static member, and use [dependency injection](https://en.wikipedia.org/wiki/Dependency_injection) similar to what is done [here](https://github.com/TylerS1066/Movecraft-Combat/blob/613c7f70e23aa47226cbd6ff7a09f3a4720a4d80/src/main/java/net/countercraft/movecraft/combat/MovecraftCombat.java#L108) with the player manager being given to the movement tracers feature.

Reply:
Just asking this is the dependency injection right?
![image](https://user-images.githubusercontent.com/68086172/163277502-7e8d7eb4-27d9-45ca-a418-27d3e9a962b3.png)
![image](https://user-images.githubusercontent.com/68086172/163277535-e49362eb-f4df-45a8-a872-0f376bdfa3ef.png)
This dependency injection is so the craftSunkByEvent is able to ask damage tracking the player initiated the obfuscation command.

Reply:
Correct, dependency injection is passing the dependency (the thing you need to access/talk to) in the constructor rather than using static access.

A slightly better example is the passing of the [`PlayerManager` to `MovementTracers`](https://github.com/TylerS1066/Movecraft-Combat/blob/e1f391c1078d919d745f1264c1c6d61a85380c2b/src/main/java/net/countercraft/movecraft/combat/MovecraftCombat.java#L108).

Reply:
Fixed